Streetcode/Streetcode.WebApi/Utils/WebParsingUtils.cs

```diff
@@ -1,73 +1,341 @@
+
+                if (deleteFile)
+                {
+                    File.Delete(excelPath);
+                }
+        }
+
+        // Following method returns name of the street optimized in such kind of way that will allow OSM Nominatim find its coordinates
+        public static string OptimizeStreetname(string streetname)
         {
-            var cellIdx = i;
-            var newRow = string.Join(",", columnsToExtract.Select(
-                c => worksheet.Cells[cellIdx, c].Value?.ToString()));
+            string streetNameStringSearchOptimized = streetname;
+
+            // OSM Nominatim does not like e.g "пров. 1-й Тихий", use "1-й Тихий провулок" instead
+            if (streetname.IndexOf("пров.") != -1)
+            {
+                streetNameStringSearchOptimized = streetname.Substring(streetname.IndexOf(" ") + 1) + " провулок";
+            }
+
+            // OSM Nominatim does not like e.g "проїзд 3-й Цукровиків", use "3-й Цукровиків проїзд" instead
+            else if (streetname.IndexOf("проїзд") != -1)
+            {
+                streetNameStringSearchOptimized = streetname.Substring(streetname.IndexOf(" ") + 1) + " проїзд";
+            }
+
+            // OSM Nominatim does not like e.g "бульв. Івана  Світличного", use "Івана  Світличного бульвар" instead
+            else if (streetname.IndexOf("бульв.") != -1)
+            {
+                streetNameStringSearchOptimized = streetname.Substring(streetname.IndexOf(" ") + 1) + " бульвар";
+            }
 
-            if (!rows.Contains(newRow))
+            // OSM Nominatim does not like e.g "шосе Стратегічне", use "Стратегічне шосе" instead
+            else if (streetname.IndexOf("шосе ") != -1)
             {
-                rows.Add(newRow);
-                stringBuilder.AppendLine(newRow);
+                streetNameStringSearchOptimized = streetname.Substring(streetname.IndexOf(" ") + 1) + " шосе";
             }
+
+            // OSM Nominatim does not like e.g "містечко Військове", use "Військове містечко" instead
+            else if (streetname.IndexOf("містечко ") != -1)
+            {
+                streetNameStringSearchOptimized = streetname.Substring(streetname.IndexOf(" ") + 1) + " містечко";
+            }
+
+            // OSM Nominatim does not like e.g "в’їзд 2-й Криничний", use "2-й Криничний в’їзд" instead
+            else if (streetname.IndexOf("в’їзд") != -1)
+            {
+                streetNameStringSearchOptimized = streetname.Substring(streetname.IndexOf(" ") + 1) + " в’їзд";
+            }
+
+            return streetNameStringSearchOptimized;
         }
 
-        await File.WriteAllTextAsync(csvPath, stringBuilder.ToString(), Encoding.UTF8);
+        // Following method calls OSM Nominatim API for single address and returns lat and lon coordinates
+        // For some addresses nothing can be returned
+        public static async Task<Tuple<string, string>> CallAPICoords(string address)
```

Comment:
- No `Async` suffix;
- Naming coule be better (smth like "FetchCoordsByAddressAsync");
- Consider uto use DocString here.

Reply:
done